src/common/peer.cc

```diff
@@ -15,146 +15,143 @@
+                                 nullptr, 0 // Service info
+                                 ,
+                                 NI_NAMEREQD // Raise an error if name resolution failed
+                                 ))
                 {
-                    if (!getnameinfo((struct sockaddr*)&sa, sizeof(sa), host, sizeof(host),
-                                     NULL, 0 // Service info
-                                     ,
-                                     NI_NAMEREQD // Raise an error if name resolution failed
-                                     ))
-                    {
-                        hostname_.assign((char*)host);
-                    }
+                    hostname_.assign(static_cast<char*>(host));
                 }
             }
-            return hostname_;
         }
+        return hostname_;
+    }
 
-        void* Peer::ssl() const { return ssl_; }
-        size_t Peer::getID() const { return id_; }
+    void* Peer::ssl() const { return ssl_; }
+    size_t Peer::getID() const { return id_; }
 
-        int Peer::fd() const
+    int Peer::fd() const
+    {
+        if (fd_ == -1)
         {
-            if (fd_ == -1)
-            {
-                throw std::runtime_error("The peer has no associated fd");
-            }
-
-            return fd_;
+            throw std::runtime_error("The peer has no associated fd");
         }
 
-        void Peer::putData(std::string name, std::shared_ptr<void> data)
-        {
-            auto it = data_.find(name);
-            if (it != std::end(data_))
-            {
-                throw std::runtime_error("The data already exists");
-            }
+        return fd_;
+    }
 
-            data_.insert(std::make_pair(std::move(name), std::move(data)));
+    void Peer::putData(std::string name, std::shared_ptr<void> data)
+    {
+        auto it = data_.find(name);
+        if (it != std::end(data_))
+        {
+            throw std::runtime_error("The data already exists");
         }
 
-        std::shared_ptr<void> Peer::getData(std::string name) const
-        {
-            auto data = tryGetData(std::move(name));
-            if (data == nullptr)
-            {
-                throw std::runtime_error("The data does not exist");
-            }
+        data_.insert(std::make_pair(std::move(name), std::move(data)));
+    }
 
-            return data;
+    std::shared_ptr<void> Peer::getData(std::string name) const
+    {
+        auto data = tryGetData(std::move(name));
+        if (data == nullptr)
+        {
+            throw std::runtime_error("The data does not exist");
         }
 
-        std::shared_ptr<void> Peer::tryGetData(std::string(name)) const
-        {
-            auto it = data_.find(name);
-            if (it == std::end(data_))
-                return nullptr;
+        return data;
+    }
 
-            return it->second;
-        }
+    std::shared_ptr<void> Peer::tryGetData(const std::string name) const
```

Comment:
Is there a good reason why `name` is being passed by value as opposed to by reference?

Reply:
I think that it was passed by value even before my changes, but I don't have access to my computer at the moment and can't look at the code

Reply:
A `string_view` makes sense to pass around by value, but an actual `string` could be expensive to copy.

Reply:
Yeah I agree, in fact I would've liked to substitute all string&s with string_views, but right now I don't have all that time.
Regarding this particular function, as you can see from the diff, the string is passed by value as it did before, and I just removed the parenthesis that made that parameter look like a constructor